cocaine12/token.go

```diff
@@ -0,0 +1,131 @@
+package cocaine12
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	tvmTokenType        = "TVM"
+	tokenRefreshTimeout = time.Second * 15
+)
+
+// Token carries information about an authorization token.
+type Token struct {
+	ty   string
+	body string
+}
+
+// Type returns token's type.
+// Depending on the context it can be empty, OAUTH, TVM etc.
+func (t *Token) Type() string {
```

Comment:
```go
func (t Token) Type() string
```
strings are immutable so Token is fast to be copied and no need to return a pointer to Token in tokenManager.Token()